src/interlocking_component/route_controller.py

```diff
@@ -17,75 +20,103 @@ def _make(self, data: dict) -> "InterlockingConfiguration":
+    as well as to notify the Interlocking faults that occur in other parts of the simulation.
+    """
 
-    def insertTrackBlocked(self, track_id: str):
+    def insert_track_blocked(self, track_id: str):
         """This method is used to block a track and recalculate the routes of relevant trains.
 
         :param track_id: the id of the blocked track
         :type track_id: str
-        :raises NotImplementedError: This has not yet been implemented.
         """
         raise NotImplementedError()
 
-    def insertTrackUnblocked(self, track_id: str):
+    def insert_track_unblocked(self, track_id: str):
         """This method is used to unblock a track and recalculate the routes of relevant trains.
 
         :param track_id: the id of the unblocked track
         :type track_id: str
-        :raises NotImplementedError: This has not yet been implemented.
         """
         raise NotImplementedError()
 
-    def insertPlatformBlocked(self, platform_id: str):
-        """This method is used to block a platform and recalculate the routes and stops of relevant trains.
+    def insert_platform_blocked(self, platform_id: str):
+        """This method is used to block a platform and recalculate the routes
+        and stops of relevant trains.
 
         :param platform_id: the id of the blocked platform
         :type platform_id: str
-        :raises NotImplementedError: This has not yet been implemented.
         """
         raise NotImplementedError()
 
-    def insertPlatformUnblocked(self, platform_id: str):
-        """This method is used to unblock a platform and recalculate the routes and stops of relevant trains.
+    def insert_platform_unblocked(self, platform_id: str):
+        """This method is used to unblock a platform and recalculate the routes
+        and stops of relevant trains.
 
         :param platform_id: the id of the unblocked platform
         :type platform_id: str
-        :raises NotImplementedError: This has not yet been implemented.
         """
         raise NotImplementedError()
 
-    def insertTrackSpeedLimitChanged(self, track_id: str):
-        """This method is used to notify the interlocking about a changed track speed limit, so that it can recalculate the routing of relevant trains.
+    def insert_track_speed_limit_changed(self, track_id: str):
+        """This method is used to notify the interlocking about a changed track speed limit,
+        so that it can recalculate the routing of relevant trains.
 
         :param track_id: the id of the track, which speedlimit changed
         :type track_id: str
-        :raises NotImplementedError: This has not yet been implemented.
         """
         raise NotImplementedError()
 
-    def insertTrainSpeedChanged(self, train_id: str):
-        """This method is used to notify the interlocking about a changed train speed limit, so that it can recalculate the routing of relevant trains.
+    def insert_train_speed_changed(self, train_id: str):
```

Comment:
```suggestion
    def insert_train_max_speed_changed(self, train: "Train"):
```
(Only the max-speed is relevant, right?)